packages/graphql-generator/src/utils/GraphQLStatementsFormatter.ts

```diff
@@ -36,7 +36,7 @@ export class GraphQLStatementsFormatter {
     this.lintOverrides = [];
     this.headerComments = [];
     this.typesPath = typesPath
-      ? typesPath.replace(/.ts/i, '')
+      ? typesPath.replace(/\.[^.]+$/, '') // remove file extensions
```

Comment:
Why not use Node's `path` module for this? `path.parse(typesPath).ext` will give you the extension to remove; `.name` will give the name exclusive of the extension, `.dir + .name` seems like what you want here.

Reply:
I thought about this and thought the regex would be cleaner. But looking at it again `path.parse` would be more error resistant.